### ISSUE TITLE
Issue with the auto correct in the Style/RescueModifier cop

### ISSUE THREAD

**Issue by filipeamoreira** [rubocop-hq/rubocop#6639]
## Expected behavior

Rubocop should correct the code so that it assigns the return value of the `rescue` block to the variable.

## Actual behavior

Rubocop corrects the code but introduces a subtle bug which may leave the variable unassigned.

## Steps to reproduce the problem

Given this line of code:

```ruby
filter_date = Date.parse(params[:filter_date]).to_date if params[:filter_date] rescue nil
```

Rubocop auto corrects this to:

```ruby
begin
  filter_date = Date.parse(params[:filter_date]).to_date if params[:filter_date]
rescue StandardError
  nil
end
```
Instead it should auto correct to:

```ruby
filter_date = begin
  Date.parse(params[:filter_date]).to_date if params[:filter_date]
rescue StandardError
  nil
end
```
Otherwise the variable will not be assigned `nil` when the rescue block is triggered. 

## RuboCop version

```
0.62.0 (using Parser 2.5.3.0, running on ruby 2.5.3 x86_64-darwin18)
```

**Comment by koic** [comment]
This is an additional information.

The following codes are the same syntax precedence.

```ruby
obj = do_something if cond rescue nil
(obj = do_something if cond) rescue nil
```

It is not below.

```ruby
obj = (do_something if cond rescue nil)
```

The result of the S-expression using Parser gem is as follows.

```console
% ruby-parse -e 'obj = do_something if cond rescue nil'
(rescue
  (if
    (send nil :cond)
    (lvasgn :obj
      (send nil :do_something)) nil)
  (resbody nil nil
    (nil)) nil)

% ruby-parse -e '(obj = do_something if cond) rescue nil'
(rescue
  (begin
    (if
      (send nil :cond)
      (lvasgn :obj
        (send nil :do_something)) nil))
  (resbody nil nil
    (nil)) nil)

% ruby-parse -e 'obj = (do_something if cond rescue nil)'
(lvasgn :obj
  (begin
    (rescue
      (if
        (send nil :cond)
        (send nil :do_something) nil)
      (resbody nil nil
        (nil)) nil)))
```

**Comment by filipeamoreira** [comment]
Thank you taking the time to verify this @koic. I was not aware of the subtle syntax differences explained in your last comment. 

I will close this issue as the bug is on our code and Rubocop is behaving correctly.
